dev/integration_tests/channels/lib/main.dart

```diff
@@ -110,11 +111,18 @@ class _TestAppState extends State<TestApp> {
     () => methodCallStandardErrorHandshake('world'),
     () => methodCallStandardNotImplementedHandshake(),
     () => basicBinaryHandshake(null),
-    () => basicBinaryHandshake(ByteData(0)),
+    if (!Platform.isMacOS)
+      // TODO(gaaclarke): This doesn't work on macOS. I looked into it and in
+      // multiple stages along the way it becomes impossible to distinguish nil
+      // from an empty buffer.
```

Comment:
Here and below: can you file a bug and link here? If the standard encoder is behaving differently on macOS from iOS, we should fix the issue.

Reply:
done

Reply:
As per the conversation with @stuartmorgan I converted these from TODO's to notes to show that the deviation is deliberate.

Reply:
What's odd is it was apparently me who introduced the different behaviour while fixing a crash, though the PR description suggests it was in an effort to behave like iOS. Will do a bit of digging to see if I can work out why.